vxquery-cli/src/main/java/org/apache/vxquery/cli/VXQuery.java

```diff
@@ -240,13 +240,16 @@ private void runQueries() throws IOException, SystemException, Exception {
      * @throws Exception
      */
     private String[] getNodeList() throws Exception {
-        Map<String, NodeControllerInfo> nodeControllerInfos = hcc.getNodeControllerInfos();
-        String[] nodeList = new String[nodeControllerInfos.size()];
-        int index = 0;
-        for (String node : nodeControllerInfos.keySet()) {
-            nodeList[index++] = node;
+        if(hcc != null) {
+            Map<String, NodeControllerInfo> nodeControllerInfos = hcc.getNodeControllerInfos();
+            String[] nodeList = new String[nodeControllerInfos.size()];
+            int index = 0;
+            for (String node : nodeControllerInfos.keySet()) {
+                nodeList[index++] = node;
+            }
+            return nodeList;
         }
-        return nodeList;
+        return new String[0];
```

Comment:
When I was using null the code was giving NPE as below : 

`Exception in thread "main" java.lang.NullPointerException 
at org.apache.vxquery.metadata.VXQueryMetadataProvider.getClusterLocations(VXQueryMetadataProvider.java:102)
	at org.apache.vxquery.metadata.VXQueryMetadataProvider.getClusterLocations(VXQueryMetadataProvider.java:97)
	at org.apache.vxquery.xmlquery.query.XMLQueryCompiler.<init>(XMLQueryCompiler.java:182)
	at org.apache.vxquery.cli.VXQuery.runQueries(VXQuery.java:264)
	at org.apache.vxquery.cli.VXQuery.execute(VXQuery.java:161)
	at org.apache.vxquery.cli.VXQuery.main(VXQuery.java:121)`